presto-main/src/main/java/com/facebook/presto/operator/scalar/annotations/ParametricScalarImplementation.java

```diff
@@ -629,6 +629,17 @@ else if (Stream.of(annotations).anyMatch(BlockPosition.class::isInstance)) {
                     }
                 }
             }
+            long valueArgumentsReturnNullOnNullCount = argumentProperties.stream()
+                    .filter(argumentProperty -> argumentProperty.getArgumentType() == VALUE_TYPE)
+                    .map(ArgumentProperty::getNullConvention)
+                    .filter(nullConvention -> nullConvention != RETURN_NULL_ON_NULL)
+                    .count();
+            checkCondition((header.getHeader().isCalledOnNullInput() && valueArgumentsReturnNullOnNullCount > 0)
```

Comment:
I personally prefer to only use `checkCondition` or `checkState` when the condition is simple. In this case, I would just explicitly write if and throw the exception :)

Reply:
Too much typing to throw an exception 😛